hoot-core-test/src/test/cpp/hoot/core/io/OsmChangeWriterTest.cpp

```diff
@@ -0,0 +1,146 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2012, 2013, 2014 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+// Hoot
+#include <hoot/core/io/ChangesetProvider.h>
+#include <hoot/core/io/ElementInputStream.h>
+#include <hoot/core/io/OsmChangeWriter.h>
+
+// Boost
+using namespace boost;
+
+// Qt
+#include <QDebug>
+#include <QFile>
+#include <QFileInfo>
+
+#include "../TestUtils.h"
+
+namespace hoot
+{
+
+//dummy implementation for testing the OsmChangeWriter
+class TestChangesetProvider : public ChangeSetProvider
+{
+
+public:
+
+  TestChangesetProvider() : _ctr(0) { }
+  ~TestChangesetProvider() { }
+
+  boost::shared_ptr<OGRSpatialReference> getProjection() const
+  {
+    return boost::shared_ptr<OGRSpatialReference>();
+  }
+
+  void close() { }
+
+  bool hasMoreChanges() { while (_ctr < 3) { return true; } return false; }
+
+  Change readNextChange()
+  {
+    NodePtr node1(new Node(Status::Unknown1, 1, 0.0, 0.0, 15.0));
+    NodePtr node2(new Node(Status::Unknown1, 2, 1.0, 0.0, 15.0));
+    WayPtr way(new Way(Status::Unknown1, 3, 15.0));
+    way->addNode(node1->getId());
+    way->addNode(node2->getId());
+    way->setTag("key1", "value1");
+    RelationPtr relation(new Relation(Status::Unknown1, 1, 15.0));
+    relation->addElement("role1", node1->getElementId());
+    relation->addElement("role2", way->getElementId());
+    relation->setTag("key2", "value2");
+
+    Change change;
+    if (_ctr == 0)
+    {
+      change.e = node1;
+      change.type = Change::Create;
+    }
+    else if (_ctr == 1)
+    {
+      change.e = relation;
+      change.type = Change::Delete;
+    }
+    else if (_ctr == 2)
+    {
+      change.e = way;
+      change.type = Change::Modify;
+    }
+    _ctr++;
+
+    return change;
+  }
+
+private:
+
+  int _ctr;
+
+};
+
+/**
+ * Not doing any tricky invalid input changeset testing here, as the ChangesetProvider should
+ * handle that.
+ */
+class OsmChangeWriterTest : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(OsmChangeWriterTest);
+    CPPUNIT_TEST(runTest);
+    CPPUNIT_TEST_SUITE_END();
+
+public:
+
+  void runTest()
+  {
+    shared_ptr<ChangeSetProvider> changesetProvider(new TestChangesetProvider());
+    OsmChangeWriter().write("test-output/OsmChangeWriterTest.osm", changesetProvider);
```

Comment:
Ha.  I actually had a dream about this last night where I realized it should be .osc instead... will make that change.